Core/Src/util.c

```diff
@@ -19,3 +21,16 @@ speed_pin_e get_speed_pin(uint16_t pin){
         return REAR_LEFT;
     }
 }
+
+//retorna uma variável do tamanho de value, porém apenas com o bit mais significativo setado em value
+uint32_t get_flag_MSB(uint32_t value) {
+    int i = 0;
+    while (value != 0) {
+        ++i;
+        value >>= 1;
+    }
+    return (1 << (i-1));
+}
```

Comment:
por mais q essa função cumpre o propósito, ela n é mt eficiente. Faz assim:
```suggestion
uint32_t get_flag_MSB(uint32_t value) {
    int flag = 1 << 31; // flag = 2^32
    while (flag != 0) {
        if (value & flag)
            return flag;
        flag >>= 1;
    }
    return 0;
}
```
(Testa se isso funciona dps rs eu fiz esse codigo agr aq no review do GitHub, talvez n funciona)

Reply:
testei numa ide de C normal e deu certo, só tive que trocar o int pra unsigned (que deveria ter feito antes)